src/LanguageServer.SemanticModel.MSBuild/MSBuildTaskMetadataCache.cs

```diff
@@ -24,10 +27,19 @@ public sealed class MSBuildTaskMetadataCache
         /// <summary>
         ///     Create a new <see cref="MSBuildTaskMetadataCache"/>.
         /// </summary>
-        public MSBuildTaskMetadataCache()
+        /// <param name="logger">
+        ///     An optional logger to use for type-reflection diagnostics.
+        /// </param>
+        public MSBuildTaskMetadataCache(ILogger logger)
         {
+            Log = logger;
         }
 
+        /// <summary>
+        ///     An optional logger to use for type-reflection diagnostics.
+        /// </summary>
+        ILogger Log { get; }
```

Comment:
I've seen this multiple times in the codebase, but I think it's better to have private elements as actual fields rather than private properties. Makes clearer separation between the two. Better to be done in a separate PR, so just want your opinion about this

Reply:
Hmm - yeah fair enough; this is another one of those deliberate choices and I get that it may seem somewhat idiosyncratic but I've just woken up so I'll try to articulate the reasoning behind it later today.

(FWIW I was thinking just yesterday about places where I've done this in the codebase at work and why so I'm going to have to remember why either way; "logging is a special case" doesn't really suffice as an explanation)

Reply:
If you don't mind, I'm going to create an issue to discuss this one, as I don't want it to hold up this PR (since it's a more global change that would be needed to keep things consistent).

Reply:
Moved to tintoy/msbuild-project-tools-server#61